test/e2e/e2e.go

```diff
@@ -192,9 +192,12 @@ var _ = ginkgo.SynchronizedBeforeSuite(func() []byte {
 	// wasting the whole run), we allow for some not-ready pods (with the
 	// number equal to the number of allowed not-ready nodes).
 	if err := framework.WaitForPodsRunningReady(c, metav1.NamespaceSystem, int32(framework.TestContext.MinStartupPods), int32(framework.TestContext.AllowedNotReadyNodes), podStartupTimeout, framework.ImagePullerLabels); err != nil {
+		framework.Logf("Error waiting for pods to be running and ready. Retrieving cluster information. Error cause is logged below after :END.")
+		framework.Logf("CLUSTER-INFORMATION:START")
```

Comment:
maybe we should instead stop dumping all of this to the log and save it into a separate file instead?

Reply:
Using multiple files is definitely a much better long-term solution. But I'm afraid this requires more knowledge than I have? Where do we put the file and how do we name it, such that it's discoverable and accessible in all the test result UIs (testgrid, gubernator, ... others?). Any idea, @kubernetes/sig-testing-feature-requests ?

Reply:
OK, I learned today that if the file is written to the artifacts directory, it will be uploaded to GCS and thus reachable through gubernator.
So I guess we could log the filename here. Initially a developer could just copy its name to clipboard, navigate to artifacts/, search for the filename and view its contents.
Evenaully we could marked up the filename here in such way that gubernator would be able to turn into a clickable hyperlink. WDYT?

Reply:
@ixdy ping?

Reply:
Sorry for delay.

You can probably use `TestContext.ReportDir`, which is set on CI runs to point to the artifacts dir. The [PrintSummaries](https://github.com/kubernetes/kubernetes/blob/9c08d30412c3e4ddfdefb7fae9a9e27d79a60eab/test/e2e/framework/util.go#L5012) function already uses this for similar functionality.

IIRC, developer runs don't have this set automatically, so you may still need to handle the case where you need to dump info to stdout (or just omit it entirely).